rfcs/062-content-api-all-search/README.md

```diff
@@ -0,0 +1,132 @@
+# Content API: All search and indexation of missing content types
+
+<!-- TODO add relevant sections -->
+
+- [Background information](#background-information)
+- ["All" index](#all-index)
+
+## Background information
+
+The current "All" search (wellcomecollection.org/search) displays separate, statically-ordered grids for Stories, Works, Images and Events. In doing so, we are unwillingly creating a hierarchy of importance between those content types which does not match their actual level of relevance. Each grid also requires its own query call, which is not efficient.
+
+As a next step, we are looking at making the "All" search expose all Prismic content types whose documents are available to our users through a UID-based URL (["Addressable content types"](#addressable-content-types)), with the results being ordered by each individual document's relevance score. Works and Images ("Collection search") will also still be available on the page, although their relevance scores will not be weighed against Addressable content types', as you can see on the image below.
+
+<img src="./assets/prototype.png" alt="Design prototype for the 'All' search page" />
+
+[Consult the design prototype here](https://www.figma.com/design/qssPpJy1lOWSFtuACajkZr/Global-search?node-id=4656-12994&node-type=canvas&m=dev).
+
+There is to be no filtering nor sorting feature on this page. Therefore, we aim to build something with minimalism in mind, allowing us to have query performance at the forefront of our concerns.
+
+We will do so by creating a new endpoint: https://api.wellcomecollection.org/content/v0/all.
+
+Its response will return [an ordered list of Addressable content types](#api-response-addressable-content-types-list), as well as [the relevant results for the Collection search](#api-response-collection-search). (**TODO: figure out what that looks like**)
```

Comment:
It might make sense for the catalogue results to be loaded from a different endpoint, and potentially for them to be loaded async on the client.

Reply:
Could/should the result be returned from the Content API then, or should it be a new endpoint built in the Catalogue API?

Reply:
It's probably worth checking in with @dana-saur, but allowing these results to be loaded async means this page can gracefully degrade, i.e. if the catalogue search is offline, content can still be searched for.

Reply:
Absolutely, I don't think she'll have any objections as it's just about being useful to the user to the extent of our capacities, but can confirm.

I thought the returned results could/should be minimal ([WIP here](https://github.com/wellcomecollection/docs/pull/112/files#diff-389a05c08a4a5c8541620f19bbab9b36b3d6561c5a0466a4f519e8adf59cac8d)), we really don't need that much, which is why I thought of a different endpoint.

Reply:
> I thought the returned results could/should be minimal

The catalogue API output when unadorned by `include` returns minimal information about works & images (though images is quite a bit more wordy than works).

Reply:
[Addressed here ](https://github.com/wellcomecollection/docs/pull/112/commits/6c44091273c59ea57372f91ebb7a853f0ec56a35)